🐛 utils: polyfill growBuffer

pietrovismara opened the pull request:
Adds `isGrowResizeSupported` to test for buffer grow support and `growBufferPolyfill`.

krispya commented on packages/utils/src/Uint32SparseSet/Uint32SparseSet.ts:
```diff
@@ -87,7 +87,12 @@ export function sparseSetRemove(sparseSet: IUint32SparseSet, value: number): voi
 
 export function sparseSetGrow(sparseSet: IUint32SparseSet, newCapacity: number): void {
-	growBuffer(sparseSet[$buffer], newCapacity);
+	if (isGrowResizeSupported()) {
+		growBuffer(sparseSet[$buffer], newCapacity);
+	} else {
+		sparseSet[$buffer] = growBufferPolyfill(sparseSet[$buffer], newCapacity);
+	}
+
```
I think polyfill should be moved into `growBuffer` so that it can be safely reused wherever we need to grow a buffer in the code base. Instead of a polyfill we should call it what it is, a different grow strategy. I don't know what you would call it, grow and replace? I'm sure there is a technical term I don't know. For testing, there should be an optional param for `growBuffer` that if true always grows and replaces, then we can test both in utils.

krispya reviewed (commented):
The over all idea is to have `growBuffer` as its own util that handles working with older browsers built-in. We can test by mocking feature detection.

pietrovismara commented:
I think I've adressed your review with the last commits.

- `growBuffer` and its polyfill are now a single function
- I added a flag param to `growBuffer` to force the grow-replace strategy
- Tested `growBuffer`
- Added the `onGrow` callback and tested it

krispya commented:
I'll review this, probably tomorrow morning.
